lib/ionic-project-transformator.ts

```diff
@@ -0,0 +1,416 @@
+				if (ionicResourceName.indexOf(folderName) >= 0) {
+					// Android resources in Ionic projects contain the resolution folder name in the resource name (e.g. drawable-hdpi-icon.png).
+					let resourceName = ionicResourceName.split(`${folderName}-`)[1];
+
+					let resourceDestinationDirectory = path.join(appBuilderAndroidResourcesDirectory, folderName, resourceName);
+
+					this.$fs.copyFile(path.join(resourceDirectory, ionicResourceName), resourceDestinationDirectory).wait();
+					return false;
+				}
+			});
+		}).future<void>()();
+	}
+
+	private checkIfPlatformIsAddedToProject(resourcesDirectory: string, platformName: string): IFuture<boolean> {
+		return Future.fromResult<boolean>(this.$fs.exists(path.join(resourcesDirectory, platformName)).wait());
+	}
+
+	private backupCurrentProject(): IFuture<void> {
+		return (() => {
+			let ionicProjectBackupDir = path.join(this.project.getProjectDir().wait(), IonicProjectTransformator.IONIC_PROJECT_BACKUP_FOLDER_NAME);
+
+			this.$logger.warn(`Creating backup in ${ionicProjectBackupDir}. This could take more than one minute. Please be patient.`);
+
+			let allProjectItems = shelljs.ls("-A", this.project.getProjectDir().wait());
+
+			if (this.$fs.exists(ionicProjectBackupDir).wait()) {
+				this.$fs.deleteDirectory(ionicProjectBackupDir).wait();
+			}
+
+			this.$fs.createDirectory(ionicProjectBackupDir).wait();
+
+			// Cannot copy directly project dir to the backup dir because it will end up in endless recursion.
+			_.each(allProjectItems, (item: string) => {
+				if (item.indexOf(IonicProjectTransformator.IONIC_PROJECT_BACKUP_FOLDER_NAME) < 0) {
+					let itemSourceDirectory = path.join(this.project.getProjectDir().wait(), item);
+					let itemDestinationDirectory = path.join(ionicProjectBackupDir, item);
+					if (this.$fs.getFsStats(itemSourceDirectory).wait().isDirectory()) {
+						shelljs.cp("-R", `${itemSourceDirectory}/`, `${itemDestinationDirectory}`);
+					} else {
+						this.$fs.copyFile(itemSourceDirectory, itemDestinationDirectory).wait();
+					}
+				}
+			});
+		}).future<void>()();
+	}
+
+	/**
+	 * Creates index.html file which will redirect to the index.html file of the Ionic project.
+	 */
+	private createReroutingIndexHtml(): IFuture<void> {
+		return (() => {
+			const indexHtmlContent = '<html><head><meta http-equiv="refresh" content="0; url=www/index.html" /></head></html>';
+			let indexHtml = path.join(this.project.getProjectDir().wait(), "index.html");
+			this.$fs.writeFile(indexHtml, indexHtmlContent).wait();
+		}).future<void>()();
+	}
+
+	private deleteEnabledPlugins(): IFuture<void> {
+		return (() => {
+			let corePlugins = _.union(
+				this.$fs.readJson(path.join(this.project.getProjectDir().wait(), ".debug.abproject")).wait().CorePlugins,
+				this.$fs.readJson(path.join(this.project.getProjectDir().wait(), ".release.abproject")).wait().CorePlugins);
+			_.each(corePlugins, plugin => {
```

Comment:
I do not like the whole approach. In case you have 10 CorePlugins, you may raise 10 exceptions...
I would prefer something like:
```TypeScript
let pluginsDir = path.join(this.project.getProjectDir().wait(), "plugins");
(this.$fs.readDirectory(pluginsDir).wait() || [])
    .filter(pl => _.contains(corePlugins, pl))
    .forEach(pl => this.$fs.deleteDirectory(path.join(pluginsDir, pl)).wait());
```